packages/common/src/util.ts

```diff
@@ -115,6 +115,46 @@ export function createId(prefix: string = "i"): string {
     .substr(2, 8)}`;
 }
 
+/**
+ * If value is not null, push it into an array, or append as a property of an object
+ * Allows for code like:
+ * ```js
+ * const vals = maybeAdd([], getProp(obj, 'some.deep.path.thatMayBe.undefined'));
+ * ```
+ *
+ * or
+ *
+ * ```js
+ * let summary = {}
+ * ['item.title', 'item.description', 'item.snippet'].forEach((k) => {
+ *   summary = maybeAdd(summary, getProp(model, k));
+ * })
```

Comment:
both of these snippets would be even _more_ useful if they were explicit about what is returned.